lib/vyasa/sangh/sheaf_lattice.ex

```diff
@@ -0,0 +1,159 @@
+defmodule Vyasa.Sangh.SheafLattice do
+  alias Vyasa.Sangh
+  alias Vyasa.Repo
+
+  @moduledoc """
+  A sheaf lattice is a flatmap that represents the tree-structure of sheafs.
+  The keys in this flatmap are the labels for a sheaf and the values are the particular sheafs themselves.
+  In this way, we can keep all of:
+  a) root sheafs: keyed by [x] where is_binary(x)
+  b) level 1 sheafs: keyed by [x, y] where is_binary(x) and is_binary(y)
+  c) level 2 sheafs: keyed by [x, y, z] where is_binary(x) and is_binary(y) and is_binary(z)
+
+  And subsequently, we can read by means of specific filters as well.
+
+  NOTE: this preloads the marks in each sheaf.
+  """
+  def create_complete_sheaf_lattice(sangh_id) when is_binary(sangh_id) do
+    root_sheafs =
+      sangh_id
+      |> Sangh.get_root_sheafs_by_session()
+      |> Enum.filter(fn s -> s.traits == ["published"] end)
+
+    [0, 1, 2]
+    |> Enum.flat_map(fn level ->
+      root_sheafs
+      |> Enum.map(fn sheaf -> to_string(sheaf.path) end)
+      |> Enum.flat_map(fn sheaf_id ->
+        Sangh.get_child_sheafs_by_session(sangh_id, sheaf_id, level)
+      end)
+      |> Enum.map(fn s ->
+        {s.path.labels,
+         s
+         |> Repo.preload(:marks)}
```

Comment:
QQ: @ks0m1c is it okay to preload marks when creating the sheaf?

for now, i'm creating a complete sheaf lattice, while ignoring the pagination piece. Will ignore pagination until things are more ready.

Reply:
should preload at get_root_sheafs_by_session/1

Reply:
done this, marking this as resolved